src/test/java/io/openliberty/tools/intellij/it/TestUtils.java

```diff
@@ -562,4 +571,41 @@ public static void copyDirectory(String sourceDirectoryLocation, String destinat
                 });
     }
 
+    /**
+     * Test to see if there has been a fatal error and JUnit should be stopped.
+     * This searches the output of this JUnit run for SocketTimeoutException which
+     * has been identified as a fatal error and occurs during the Mac tests.
+     * It is set up as a static reader so that we do not reread the whole file
+     * after each test.
+     */
+    final static String outputFile = System.getenv("JUNIT_OUTPUT_TXT");
+    final static BufferedReader reader;
+
+    static {
+        try {
+            reader = new BufferedReader(new InputStreamReader(new FileInputStream(outputFile)));
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void detectFatalError() {
```

Comment:
synchronized? The reader has state and might break if more than one thread comes in here at a time.

Reply:
Yes, that sounds good.